maestro-orchestra/src/main/java/maestro/orchestra/Orchestra.kt

```diff
@@ -113,6 +113,45 @@ class Orchestra(
 
     private val rawCommandToMetadata = mutableMapOf<MaestroCommand, CommandMetadata>()
 
+    private var isStopping = false
+    private var isPaused = false
+    private val pauseLock = Object()
+
+    fun stopFlow() {
+        isStopping = true
```

Comment:
Can we use coroutine to manage these states? This is something that was already started here:

https://github.com/mobile-dev-inc/Maestro/pull/2488/files

I feel stopped is probably already implemented there. Since when coroutine is cancelled and coroutine becomes inactive we directly return from the function here:

https://github.com/mobile-dev-inc/Maestro/pull/2488/files#diff-00989d49f0eaa1f9e614580322e3b2ec79d9915a804ffa004e2f53ba36215698R188